### PR TITLE
Document project API endpoints

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: docs/index.rst [hunk 1]
```diff
@@ -241,6 +272,186 @@ Dialogue Summaries
...
+       "is_archived": false,
+       "is_published": false,
+       "has_changes": false
+     }]
+   }
+
+
+Archiving projects
+~~~~~~~~~~~~~~~~~~
+A project can be archived by setting ``is_archived`` to ``true`` when :ref:`replacing <projects-put>` or :ref:`partially updating <projects-patch>` the project description. The project is still accessible via the api, but its dialogues will no longer be triggered by any events that were previously associated to them.
```
Comment (justinvdm) [inline docs/index.rst]: A bit unsure about this, still wondering if there is something better we can do for archived projects and dialogues. This is the best I can think of for now, but will give this some more thought.

**Comment by justinvdm** [comment]
Now depends on #15, otherwise ready for review.

File: docs/index.rst [hunk 2]
```diff
@@ -241,6 +272,186 @@ Dialogue Summaries
...
+
+.. http:post:: /projects/
+
+  Creates a new project with the :ref:`project description <data-projects>`
+  given in the request body and returns the created projects's description,
+  along with the generated dialogue ``id`` field and ``url`` field for
+  accessing the project description.
+
+  .. TODO mention that the user creating the project is given admin access once
+  we have documented permission levels
```
Comment (hodgestar) [inline docs/index.rst]: Do we want to add this now that #11 is done?

File: docs/index.rst (hunk 1 above)
Comment (hodgestar) [inline docs/index.rst]: We probably also don't want a project to modified after it is archived. Perhaps we should leave out archiving for now and document it once we're ready?

**Comment by hodgestar** [comment]
Left two small comments, otherwise looks good.

File: docs/index.rst (hunk 1 above)
Comment (justinvdm) [inline docs/index.rst]: Sounds good.

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**